health/health.c

```diff
@@ -45,12 +45,14 @@ inline char *health_stock_config_dir(void) {
  * Function used to initialize the silencer structure.
  */
 void health_silencers_init(void) {
-    struct stat statbuf;
-    if (!stat(silencers_filename,&statbuf)) {
-        off_t length = statbuf.st_size;
-        if (length && length < HEALTH_SILENCERS_MAX_FILE_LEN) {
-            FILE *fd = fopen(silencers_filename, "r");
-            if (fd) {
+    if (silencers_filename) {
```

Comment:
By looking at the configuration code `silencers_filename` is never going to be NULL, but even if it could be it would be nice to have an error print for that case.

Reply:
You are correct, this is a global variable that will never be NULL, because we give a default value for it when the user does not specify. Thanks!

Reply:
Hmm, yes, that's why the original code of Kosta was not doing any error handling, some file descriptors in the range are bound to not exist.

See here:
https://github.com/netdata/netdata/blob/4f64e8edbdb0d4b68b882aa34474a0156b6ba150/libnetdata/popen/popen.c#L73

Reply:
Yes, let us hope that the `(void)` will be enough to coverity does not raise more errors.